lock-api/src/main/java/com/palantir/lock/client/RemoteTimelockServiceAdapter.java

```diff
@@ -173,20 +195,20 @@ public long currentTimeMillis() {
 
     @Override
     public TimestampLeaseResults acquireTimestampLeases(Map<TimestampLeaseName, Integer> requests) {
-        // TODO(aalouane): implement!
-        throw new UnsupportedOperationException("Not implemented yet!");
+        return timestampLeaseAcquirer.acquireNamedTimestampLeases(requests);
     }
 
     @Override
     public Map<TimestampLeaseName, Long> getMinLeasedTimestamps(Set<TimestampLeaseName> timestampNames) {
-        // TODO(aalouane): implement!
-        throw new UnsupportedOperationException("Not implemented yet!");
+        return minLeasedTimestampGetter.getMinLeasedTimestamps(timestampNames);
     }
```

Comment:
Finishing this up this way, now I realize there is an alternative way for the unlocking:
- have the `TimestampLeaseAcquirer` return a `result|err` type where the `err` contains anything that needs to be unlocked
- do the unlocking in this class
It's possible for this to be annoying in that we may lose context/stacktrace, but we can pass any as needed